barretenberg/cpp/src/barretenberg/benchmark/widgets_bench/widget.bench.cpp

```diff
@@ -36,12 +58,46 @@ template <typename Flavor, typename Widget> void execute_widget(::benchmark::Sta
         widget.compute_quotient_contribution(barretenberg::fr::random_element(), data.transcript);
     }
 }
-void plookup_auxiliary_kernel(::benchmark::State& state) noexcept
+
+template <typename Widget> void quotient_contribution(::benchmark::State& state) noexcept
 {
     BasicPlonkKeyAndTranscript data = get_plonk_key_and_transcript();
+    Widget widget(data.key.get());
+    for (auto _ : state) {
+#ifdef GET_PER_ROW_TIME
+        auto start = std::chrono::high_resolution_clock::now();
+#endif
+        widget.compute_quotient_contribution(barretenberg::fr::random_element(), data.transcript);
+#ifdef GET_PER_ROW_TIME
+        auto end = std::chrono::high_resolution_clock::now();
+        auto elapsed_seconds = std::chrono::duration_cast<std::chrono::duration<double>>(end - start);
+        state.SetIterationTime(elapsed_seconds.count() / (LARGE_DOMAIN_SIZE * WIDGET_BENCH_TEST_CIRCUIT_SIZE));
+#endif
+    }
+}
 
-    using FFTGetter = ProverPlookupAuxiliaryWidget<ultra_settings>::FFTGetter;
-    using FFTKernel = ProverPlookupAuxiliaryWidget<ultra_settings>::FFTKernel;
+#ifdef GET_PER_ROW_TIME
+BENCHMARK(quotient_contribution<ProverPlookupArithmeticWidget<ultra_settings>>)->Iterations(1)->UseManualTime();
+BENCHMARK(quotient_contribution<ProverGenPermSortWidget<ultra_settings>>)->Iterations(1)->UseManualTime();
+BENCHMARK(quotient_contribution<ProverEllipticWidget<ultra_settings>>)->Iterations(1)->UseManualTime();
+BENCHMARK(quotient_contribution<ProverPlookupAuxiliaryWidget<ultra_settings>>)->Iterations(1)->UseManualTime();
+BENCHMARK(quotient_contribution<ProverPlookupWidget<4>>)->Iterations(1)->UseManualTime();
+BENCHMARK(quotient_contribution<ProverPermutationWidget<4, true>>)->Iterations(1)->UseManualTime();
+#else
+BENCHMARK(quotient_contribution<ProverPlookupArithmeticWidget<ultra_settings>>)->Iterations(1);
+BENCHMARK(quotient_contribution<ProverGenPermSortWidget<ultra_settings>>)->Iterations(1);
+BENCHMARK(quotient_contribution<ProverEllipticWidget<ultra_settings>>)->Iterations(1);
+BENCHMARK(quotient_contribution<ProverPlookupAuxiliaryWidget<ultra_settings>>)->Iterations(1);
+BENCHMARK(quotient_contribution<ProverPlookupWidget<4>>)->Iterations(1);
+BENCHMARK(quotient_contribution<ProverPermutationWidget<4, true>>)->Iterations(1);
+#endif
+
+template <typename Widget> void accumulate_contribution(::benchmark::State& state) noexcept
+{
+    BasicPlonkKeyAndTranscript data = get_plonk_key_and_transcript();
+
+    using FFTGetter = typename Widget::FFTGetter;
+    using FFTKernel = typename Widget::FFTKernel;
```

Comment:
This isn't new to this PR but the (hidden) comment below: 
"// NOTE: this simply calls the following 3 functions it does NOT try to replicate ProverPlookupAuxiliaryWidget
 // logic exactly" 
is now outdated

Reply:
Removed